### ISSUE TITLE
False positive on static access of class constants

### ISSUE THREAD

**Issue by duncancumming** [phpmd/phpmd#550]
Trying to use class constants give me this PHPMD warning: _"avoid using static access to class 'self'"_

e.g.

```
class Foo {
  const MAX = 10;

  public function isMax($x)
  {
    return $x === self::MAX;
  }
}
```

[The documentation](https://phpmd.org/rules/cleancode.html#staticaccess) states _"The only case when static access is acceptable is when used for factory methods"_.  

I'm not sure what's wrong with using constants like this.  

See also the workaround here, can this be implemented?
- https://stackoverflow.com/questions/20552282/phpmd-avoid-using-static-access-to-class-using-a-constant
